pandas/_libs/tslibs/src/datetime/np_datetime_strings.c

```diff
@@ -235,13 +271,17 @@ int parse_iso_8601_datetime(const char *str, int len, int want_exc,
         if (out_local != NULL) {
             *out_local = 0;
         }
+        if (format_len) {
+          goto parse_error;
+        }
```

Comment:
A question: do we want the block after `parse_error` to also reflect this format introduction with a meaningful error message? If format is incorrect, the error message will be about the date string, not the format string...

Reply:
That's a good question - the error message will still mention the format though, e.g.:
```python
>>> import pandas as pd
>>> pd.to_datetime('2022-01-01', format='%Y-%m-%d %H')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/megorelli/pandas-dev/pandas/core/tools/datetimes.py", line 1132, in to_datetime
    result = convert_listlike(np.array([arg]), format)[0]
  File "/home/megorelli/pandas-dev/pandas/core/tools/datetimes.py", line 441, in _convert_listlike_datetimes
    result, tz_parsed = objects_to_datetime64ns(
  File "/home/megorelli/pandas-dev/pandas/core/arrays/datetimes.py", line 2205, in objects_to_datetime64ns
    result, tz_parsed = tslib.array_to_datetime(
  File "pandas/_libs/tslib.pyx", line 440, in pandas._libs.tslib.array_to_datetime
    cpdef array_to_datetime(
  File "pandas/_libs/tslib.pyx", line 622, in pandas._libs.tslib.array_to_datetime
    raise ValueError(
ValueError: time data "2022-01-01" at position 0 doesn't match format "%Y-%m-%d %H"
```